Shopify/shopify-cli
# Fix `theme serve` raised error when there is a permissions error

### WHY are these changes introduced?

Potentially addresses #2059, which seems to be where user is logged in but they have no theme edit permissions (or otherwise aren't correctly authenticated)

After updates in #2020 (https://github.com/Shopify/shopify-cli/commit/f3c3e78bb52dd214fbe5fe5738de443951e11015#diff-3cdbab84dd5036fc4a850add4e66246deec39779563f9bd7c5b2bff0559e6e3cR31), there is no `theme` object to access if there is a raised permissions error, so `theme.shop` raises an error in the rescue block (https://github.com/Shopify/shopify-cli/blob/main/lib/shopify_cli/theme/dev_server.rb#L83)

See also #2071

### WHAT is this pull request doing?

This PR creates a new error handling method (`Common::ErrorHelper`, following @karreiro pattern in #2066) to create a new generic theme to get access to the shop name for the error message.

Additionally I noticed that the other theme commands in this same case were raising 403 errors rather than showing a correct error message, so this PR uses the same method to handle those errors as well.

#### notes/caveats
It's a little awkward including this same ErrorHelper module in with the DevServer, so I'm happy to do something different if desired.

I potentially thought it might be simpler to rescue the permissions errors in the DevelopmentTheme `find` and `find_or_create!` methods, but there are a couple of problems there:
- `find` doesn't raise an error if the theme doesn't exist, since `exists?` returns false if there is no ID without checking the API
- the other commands (`pull`/`push` etc) may have other API requests that will raise the same errors, so would need the same error handling anyhow.

### How to test your changes?
- be logged in to a store with no permissions
- run `SHOPIFY_CLI_STACKTRACE=1 bundle exec shopify theme serve`
  - see `You are not authorized to edit themes on...` error message instead of NoMethodError
- run `SHOPIFY_CLI_STACKTRACE=1 bundle exec shopify theme pull `
  - see `You are not authorized to edit themes on...` error message instead of 403 Error

### Update checklist
- [ ] Will update specs as needed once the general direction is approved.
